src/Microsoft.AspNet.Mvc.Razor/Precompilation/SymbolUtilities.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+#if DNXCORE50
+using System.Reflection;
+#endif
+using Microsoft.CodeAnalysis;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.Razor.Precompilation
+{
+    public static class SymbolUtilities
+    {
+        public static IEnumerable<TAttribute> GetCustomAttributes<TAttribute>(ISymbol symbol) where TAttribute : Attribute
+        {
+            var attributes = symbol.GetAttributes();
+            var attributeFullName = typeof(TAttribute).FullName;
+            return attributes
+                .Where(
+                    attribute => string.Equals(GetFullName(attribute.AttributeClass), attributeFullName, StringComparison.Ordinal))
+                .Select(a => CreateAttribute<TAttribute>(a));
+        }
+
+        public static string GetFullName(ITypeSymbol typeSymbol)
+        {
+            var nameBuilder = new StringBuilder(typeSymbol.Name);
```

Comment:
Talked in person about robustness of this method (doesn't work for every scenario) and it's expected to not be 100% fool proof.